test/gurka/test_time_dependent_tags.cc

```diff
@@ -0,0 +1,115 @@
+        {"DE", {{"highway", "motorway"}}}, {"EC", {{"highway", "motorway"}}},
+    };
+    layout = gurka::detail::map_to_coordinates(ascii_map, 10);
+    restriction_members = {
+        {gurka::way_member, "AB", "from"},
+        {gurka::way_member, "BC", "to"},
+        {gurka::node_member, "B", "via"},
+    };
+
+    working_hours = {
+        "2021-04-22T14:00",
+        "2021-04-22T23:00",
+        "2021-04-25T14:00",
+        "2021-04-25T23:00",
+    };
+  }
+};
+gurka::ways TimeDependentTags::ways = {};
+gurka::nodelayout TimeDependentTags::layout = {};
+std::vector<gurka::relation_member> TimeDependentTags::restriction_members = {};
+std::vector<std::string> TimeDependentTags::working_hours = {};
+
+TEST_F(TimeDependentTags, HourRestrictions) {
+  tags hour_only = {
+      {"type", "restriction"},
+      {"restriction", "no_left_turn"},
+      {"hour_on", "07:00"},
+      {"hour_off", "19:00"},
+  };
+  const gurka::relations restrictions = {{restriction_members, hour_only}};
+  const gurka::map map =
+      gurka::buildtiles(layout, ways, {}, restrictions,
+                        "test/data/gurka_parse_time_dependent_restrictions_hour",
+                        {{"mjolnir.timezone", VALHALLA_BUILD_DIR "test/data/tz.sqlite"}});
+
+  for (int x = 0; x < working_hours.size(); ++x) {
+    auto result =
+        gurka::do_action(valhalla::Options::route, map, {"A", "F"}, "auto",
+                         {{"/date_time/type", "1"}, {"/date_time/value", working_hours.at(x)}});
+    if (x == 0 || x == 2) { //  Restriction is on.
+      gurka::assert::raw::expect_path(result, {"AB", "BD", "DE", "EC", "CF"});
+    } else if (x == 1 || x == 3) { //  Restriction is off.
+      gurka::assert::raw::expect_path(result, {"AB", "BC", "CF"});
+    }
+  }
+}
+
+TEST_F(TimeDependentTags, DayAndHourRestrictions) {
+  tags day_hour = {{"type", "restriction"}, {"restriction", "no_left_turn"},
+                   {"day_on", "Monday"},    {"day_off", "Friday"},
+                   {"hour_on", "07:00"},    {"hour_off", "19:00"}};
+  const gurka::relations restrictions = {{restriction_members, day_hour}};
+  const gurka::map map =
+      gurka::buildtiles(layout, ways, {}, restrictions,
+                        "test/data/gurka_time_dependent_restrictions_day_and_hour",
+                        {{"mjolnir.timezone", VALHALLA_BUILD_DIR "test/data/tz.sqlite"}});
+
+  for (int x = 0; x < working_hours.size(); ++x) {
+    auto result =
+        gurka::do_action(valhalla::Options::route, map, {"A", "F"}, "auto",
+                         {{"/date_time/type", "1"}, {"/date_time/value", working_hours.at(x)}});
+    if (x == 0) { //  Restriction is on.
+      gurka::assert::raw::expect_path(result, {"AB", "BD", "DE", "EC", "CF"});
+    } else { //  Restriction is off.
+      gurka::assert::raw::expect_path(result, {"AB", "BC", "CF"});
+    }
+  }
+}
+
+TEST_F(TimeDependentTags, IgnoreDayOffOnlyRestriction) {
+  tags day_off = {
```

Comment:
if I understood correctly having only `day_on` or `hour_off/hour_on` also should be ignored. could you add all these cases ?